app/assets/javascripts/discourse/app/instance-initializers/sidebar.js

```diff
@@ -0,0 +1,29 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import I18n from "I18n";
+
+export default {
+  initialize() {
+    withPluginApi("1.7.1", (api) => {
+      api.addSidebarPanel((BaseCustomSidebarPanel) => {
+        const MainSidebarPanel = class extends BaseCustomSidebarPanel {
+          get key() {
+            return "main";
```

Comment:
I'm quite confused why we need this here where we're using a plugin API in core. If we're already in core code land, we can just create the object directly and do not need to depend on the plugin API.